piff/optical_model.py

```
@@ -175,52 +256,215 @@ def fit(self, star):
-                atm = galsim.VonKarman(**self.atm_kwargs)
-            else:
-                atm = galsim.Kolmogorov(**self.atm_kwargs)
-            prof.append(atm)
 
-        # optics
-        if params is None or len(params) == 0:
-            # no aberrations.  Just the basic opt_kwargs
-            optics = galsim.OpticalPSF(**self.opt_kwargs)
+        if self.mirror_figure_screen:
+
+            optical_screen = galsim.OpticalScreen(diam=self.diam, aberrations=zernike_coeff, lam_0=self.lam)
+            screens = galsim.PhaseScreenList([optical_screen, self.mirror_figure_screen])
+            optics_psf = galsim.PhaseScreenPSF(screen_list=screens, aper=self.aperture, lam=self.lam, gsparams=self.gsparams
+            )
+
         else:
-            aberrations = [0,0,0,0] + list(params)
-            optics = galsim.OpticalPSF(aberrations=aberrations, **self.opt_kwargs)
+            optics_psf = galsim.OpticalPSF(
+                lam=self.lam, diam=self.diam, aper=self.aperture,
+                aberrations=zernike_coeff, gsparams=self.gsparams
+            )
+
+        return optics_psf
 
-        # convolve together
-        prof.append(optics)
+    @lru_cache(maxsize=512)
+    def getAtmosphere(self, r0, L0=None, g1=0.0, g2=0.0):
+        """ Get the Atmospheric PSF component.
 
-        if len(prof) == 1:
-            prof = prof[0]
+        :param r0:          Fried parameter
+        :param L0:          Outer scale
+        :param g1:          Shear g1 component
+        :param g2:          Shear g2 component
+
+        :returns: a galsim.GSObject instance
+        """
+        if self.atmo_type == 'VonKarman':
+            if L0 is None or L0==0.:
+                raise ValueError("No value specified for L0 ")
+            try:
+                atm = galsim.VonKarman(lam=self.lam, r0=r0, L0=L0, flux=1.0, gsparams=self.gsparams)
+            except:
+                self.logger.error("Problem making a VonKarman r0,L0 = %f,%f" % (r0,L0))
```

I'm not sure this is right.  Do we really want to catch and ignore (albeit with a logger message) when VonKarmann raises an error?  This is going to lead to another error later when this function tries to return `atm`.  Probably better to just let whatever error galsim raises through so the user can see it rather than swallow it here.
(Also, btw, it's bad form to just use `except` with no types.  When catching errors you should list the errors that you are expecting and know how to handle.  Some like `KeyboardInterrupt` should never be caught.)

---

ok - removed this try, except